src/main/java/com/exasol/bucketfs/uploadnecassity/ChecksumUploadNecessityCheckStrategy.java

```diff
@@ -0,0 +1,144 @@
+package com.exasol.bucketfs.uploadnecassity;
+
+import static com.exasol.bucketfs.uploadnecassity.ByteArrayToHexConverter.toHex;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.*;
+import java.sql.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import com.exasol.bucketfs.*;
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * This {@link UploadNecessityCheckStrategy} compares the checksum of the local file and the file on BucketFs and only
+ * uploads the file if they differ or if the file does not exist in BucketFs.
+ */
+//[impl->dsn~conditional-upload~1]
+public class ChecksumUploadNecessityCheckStrategy implements UploadNecessityCheckStrategy {
+    private static final String UDF_SCHEMA = "BUCKET_FS_JAVA_HELPER";
+    private static final String UDF_NAME = "bucketfs_md5_sum";
+    private static final String UDF_FULL_NAME = UDF_SCHEMA + "." + UDF_NAME;
+    private static final int ONE_MEGABYTE = 1000000;
+    private final Connection sqlConnection;
+
+    /**
+     * Create a new instance of {@link ChecksumUploadNecessityCheckStrategy}.
+     * 
+     * @param sqlConnection SQL connection to the Exasol database. (used for calculating the checksum of the file in
+     *                      BucketFs in a UDF)
+     */
+    public ChecksumUploadNecessityCheckStrategy(final Connection sqlConnection) {
+        this.sqlConnection = sqlConnection;
+    }
+
+    @Override
+    public boolean isUploadNecessary(final Path file, final String fullFileNameInBucketFs, final ReadOnlyBucket bucket)
+            throws BucketAccessException {
+        try {
+            final String[] parts = fullFileNameInBucketFs.split(BucketConstants.PATH_SEPARATOR);
```

Comment:
Use the `Path` class instead. It has those features built in.

Reply:
Path is platform dependent. On Windows it will lead to wrong results